File: api/demo/src/main/java/com/example/demo/entity/ExtendedField.java
package com.example.demo.entity;

import javax.persistence.*;
import java.util.List;

/**
 * 扩展字段定义
 *
 * @author panjiaqi
 */
@Entity
public class ExtendedField implements BaseEntity<Long> {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 名称
     */
    private String name;
    /**
     * 扩展字段记录
     */
    @OneToMany(mappedBy = "extendedField")
    private List<GoodExtendedField> goodExtendedFieldList;

    public ExtendedField() {
    }

    @Override
    public Long getId() {
        return id;
    }

    @Override
    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<GoodExtendedField> getGoodExtendedFieldList() {
        return goodExtendedFieldList;
    }

    public void setGoodExtendedFieldList(List<GoodExtendedField> goodExtendedFieldList) {
        this.goodExtendedFieldList = goodExtendedFieldList;
    }
}

File: web/src/app/common/extended-field.ts
import { GoodExtendedField } from './good-extended-field';

/**
 * 扩展字段定义
 */
export class ExtendedField {
  /** id */
  id: number;

  /** 名称 */
  name: string;

  /** 扩展字段记录 */
  goodExtendedFieldList: Array<GoodExtendedField>;

  constructor(data?: { id?: number, name?: string, goodExtendedFieldList?: Array<GoodExtendedField> }) {
    if (data) {
      if (data.id) {
        this.id = data.id;
      }

      if (data.name) {
        this.name = data.name;
      }

      if (data.goodExtendedFieldList) {
        this.goodExtendedFieldList = data.goodExtendedFieldList;
      }
    }
  }
}
